elements/rh-cta/src/rh-cta.scss

```diff
@@ -1,155 +1,367 @@
+$rh-cta--default-bg-color--solid--hover:  darken($rh-global--color--gray-batman, 10%);
+$rh-cta--default-text-color--solid--hover:$rh-global--color--white;
+
+$rh-cta--primary-bg-color--solid:         $rh-global--color--blue-azure;
+$rh-cta--primary-text-color--solid:       $rh-global--color--white;
+$rh-cta--primary-bg-color--solid--hover:  darken($rh-global--color--blue-azure, 10%);
+$rh-cta--primary-text-color--solid--hover:$rh-global--color--white;
+
+$rh-cta--secondary-bg-color--solid:       $rh-global--color--gray-iron;
+$rh-cta--secondary-text-color--solid:     $rh-global--color--white;
+$rh-cta--secondary-bg-color--solid--hover:darken($rh-global--color--gray-iron, 10%);
+$rh-cta--secondary-text-color--solid--hover:$rh-global--color--white;
+
+$rh-cta--accent-bg-color--solid:          $rh-global--color--red;
+$rh-cta--accent-text-color--solid:        $rh-global--color--white;
+$rh-cta--accent-bg-color--solid--hover:   darken($rh-global--color--red, 10%);
+$rh-cta--accent-text-color--solid--hover: $rh-global--color--white;
+
+
 
-@mixin make-cta-arrow($arrow-color: $rh-global--link-color) {
-  margin-left: var(--rhe-theme--spacer--xs, $rh-global--spacer--xs);
-  vertical-align: middle;
-  @include rh-arrow(right, $arrow-color, 0.313em, 0.313em, true);
-  position: relative; // temp
-  display: inline-block; // temp
+/// ===========================================================================
+/// Mixins
+/// ===========================================================================
+
+/// Make Default CTA State
+/// Creates the visited, hover, and focus states and colors its arrow
+/// @param {String} $this-name - State name
+/// @param {String} $this-color - Color
+@mixin make-default-cta-state($this-name: "hover", $this-color: rh-var("link-color--hover")) {
+  & ::slotted(a:#{$this-name}) {
+    color: $this-color;
+
+    &::after {
+      @include rh-arrow-color($this-color);
+    }
+  }
 }
 
-@mixin make-default-cta-color($color-name: white, $color: $rh-global-color--white) {
-  @at-root :host(.#{$color-name}) {
+/// Make Default CTA Variation
+/// Creates color variations of the default CTA
+/// @param {String} $this-selector - Selector query
+/// @param {String} $this-name - Name to be used in Var set up
+/// @param {String} $this-color - Fallback color
+/// @param {String} $this-color-hover - Fallback hover color
+/// @param {String} $this-color-focus - Fallback focus color
+/// @param {String} $this-color-visited - Fallback visited color
+/// @param {String} $this-modifier - Modifier to be used in Var set up, like "--inverted"
+@mixin make-default-cta-variation(
+    $this-selector: ".primary",
+    $this-name: "primary",
+    $this-color: $rh-cta--default-color,
+    $this-color-hover: $rh-cta--default-color--hover,
+    $this-color-focus: $rh-cta--default-color--focus,
+    $this-color-visited: $rh-cta--default-color--visited,
+    $this-modifier: ""
+  ) {
+
```

Comment:
```scss
$cta-settings: (
     primary-link-color: primary-link-color, $rh-cta--default-color,
     primary-link-color--visited: primary-link-color--visited, $rh-cta--default-color--visited,
     primary-link-color--hover: primary-link-color--hover, $rh-cta--default-color--hover,
     primary-link-color--focus: primary-link-color--focus, $rh-cta--default-color--focus
);
```

^ the map contains the variable name as the key with a value that is a list containing the theme value and the fallback color.  If only one value exists in the list, the mixin will still generate the right variable listing.

Reply:
This example looks a bit redundant with the theme names being the same but there will be a lot of examples where the theme name might not resemble the variable name at all.

Reply:
Let's meet up next week to work on this idea.